python-avd/pyavd/_eos_cli_config_gen/schema/schema_fragments/ethernet_interfaces.schema.yml

```diff
@@ -1945,6 +1945,200 @@ keys:
+                      items:
+                        type: str
+                    interfaces:
+                      description: Interfaces like -  Ethernet, InternalRecirc, Port-Channel, Recirc-Channel.
+                      type: list
+                      items:
+                        type: str
+                    nexthop_groups:
+                      type: list
+                      description: Default nexthop-group names.
+                      items:
+                        type: str
+                identity:
+                  type: dict
+                  keys:
+                    id:
+                      type: int
+                      convert_types:
+                        - str
+                      min: 1
+                      max: 65535
+                      description: Tap port VLAN ID (1-4094) or DzGRE extended ID (1-65535).
+                    inner_vlan:
+                      description: Tap port inner VLAN ID. Only applicable if `id` is a VLAN ID (1-4094).
+                      type: int
+                      convert_types:
+                        - str
+                      min: 1
+                      max: 4094
+                mpls_pop_all:
+                  type: bool
+                  description: Pop all MPLS labels.
+                native_vlan:
+                  type: int
+                  convert_types:
+                    - str
+                  min: 1
+                  max: 4094
+                  description: Native VLAN ID when interface is in tap mode.
+                truncation:
+                  type: dict
+                  keys:
+                    enabled:
+                      type: bool
+                      required: true
+                    size:
+                      description: Ingress packet truncation size in bytes.
+                      type: int
+                      convert_types:
+                        - str
+                      min: 100
+                      max: 9236
+                mac_address:
+                  type: dict
+                  keys:
+                    src:
+                      type: str
+                      pattern: "^([0-9a-f]{2}:){5}[0-9a-f]{2}$"
+                      description: MAC address for the source.
+                    dest:
+                      type: str
+                      required: true
+                      pattern: "^([0-9a-f]{2}:){5}[0-9a-f]{2}$"
+                      description: MAC address for the destination.
+                encapsulation:
+                  type: dict
+                  keys:
+                    vxlan_strip:
+                      type: bool
+                      description: |-
+                        Strip VXLAN encapsulation header.
+                        `encapsulation.vxlan_strip` and `mpls_pop_all` are mutually exclusive.
+                        `mpls_pop_all` takes precedence.
+                    gre:
+                      type: list
```

Comment:
could any of the fields be a primary_key?

Reply:
No, there could be commands like this -
```
   switchport tap encapsulation gre protocol 0x3 strip
   switchport tap encapsulation gre strip
   switchport tap encapsulation gre destination 1.1.1.1 protocol 0x0 strip
   switchport tap encapsulation gre destination 1.1.1.2 protocol 0xffff strip
   switchport tap encapsulation gre destination 1.1.1.3 protocol 0x2 strip
   switchport tap encapsulation gre destination 1.1.1.4 protocol 0x1 strip
```

Reply:
Refactored code to fix ordering!

Reply:
Maybe we should adjust the data model to fit this better. Add the protocol and strip bool on the gre level and then a `destination` list with primary key destination and the same options again?

Reply:
`protocol` can not be a boolean, it needs a value (also at gre level)
and could be multiple commands like -
  ```
   switchport tap encapsulation gre protocol 0x0001 strip
   switchport tap encapsulation gre protocol 0x0002 feature header length 3 strip
   switchport tap encapsulation gre protocol 0x0003 feature header length 2 strip re-encapsulation ethernet
```

Reply:
Created the new model for GRE